scipy/stats/_stats_py.py

```diff
@@ -3956,31 +3957,157 @@ def __init__(self, msg=None):
         self.args = (msg,)
 
 
-def pearsonr(x, y):
+def _pearsonr_fisher_ci(r, n, confidence_level, alternative):
+    """
+    Compute the confidence interval for Pearson's R.
+
+    Fisher's transformation is used to compute the confidence interval
+    (https://en.wikipedia.org/wiki/Fisher_transformation).
+    """
+    if r == 1:
+        zr = np.inf
+    elif r == -1:
+        zr = -np.inf
+    else:
+        zr = np.arctanh(r)
+    se = np.sqrt(1 / (n - 3))
```

Comment:
Might cause a zero-division error:

```python
>>> from scipy.stats import pearsonr
>>> 
>>> res = pearsonr([1,2,3], [4,5,6])
>>> res.confidence_interval()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/tirthasheshpatel/tensor_libs/scipy/build-2-install/lib/python3.9/site-packages/scipy/stats/_stats_py.py", line 4054, in confidence_interval
    return _pearsonr_fisher_ci(self.statistic, self._n, confidence_level,
  File "/home/tirthasheshpatel/tensor_libs/scipy/build-2-install/lib/python3.9/site-packages/scipy/stats/_stats_py.py", line 3973, in _pearsonr_fisher_ci
    se = np.sqrt(1 / (n - 3))
ZeroDivisionError: division by zero
```

```suggestion
    if n == 2:
        se = np.nan
    elif n == 3:
        se = np.inf
    else:
        se = np.sqrt(1 / (n - 3))
```

Reply:
Thanks @tirthasheshpatel. Looks appropriate.